Ansible updates

eparis opened the pull request:
Supports certs, kube-system namespace, additional vagrant providers and cleanups.

jeffbean committed:
Ansible: Adding variable for kubernetes addon dir

jeffbean committed:
Ansible: Adding variable for kubernetes apiserver port

jeffbean committed:
Ansible: Updating README

eparis committed:
Ansible: cert gen: use inventory hostname not ip address

eparis committed:
Ansible: Vagrant: Update to os1 vagrant scripts

Do the /etc/host creation with vagrant, so it uses internal instead of
external ips (hostmanager only knew about the public ip)

Ignore errors on docker failure when 'restarting' docker in flannel
handler. If this is a clean install, we haven't run 'node' yet so docker
isn't installed so it doesn't need to be started. It would be better to
be more specific in ignoring errors though...

eparis committed:
Ansible: Support new upstream non-default namespace for addons

eparis committed:
Ansible: Minor typo in play text (no functional change)

eparis committed:
Ansible: Run from locally built binaries

Instead of from distro packages

eparis committed:
Ansible: Vagrant: Add vbox support

eparis committed:
Ansible: Do not fail flannel restart is docker not installed

On first install flannel runs before docker is installed. So do not fail
if docker is not installed

eparis committed:
Ansible: Vagrant: Add libvirt support

eparis committed:
Ansible: vagrant: generic set_provider() function

eparis commented:
@jeffbean @jayunit100
 
Please ok your commits

jeffbean commented on contrib/ansible/vagrant/README.md:
```diff
@@ -0,0 +1,49 @@
+
+## Before you start !
+
+You will need to pick your vagrant provider. Currently supported are openstack, libvirt, and virtualbox.
```
Can we add how to pick. Once I have picked do I have to take any actions? How do I configure that provider?

eparis replied:
I'll add a not here.  vagrant tries to just be smart.

jeffbean commented on contrib/ansible/vagrant/openstack_config.yml:
```diff
@@ -0,0 +1,12 @@
+os_username: eparis
```
Do you want to commit this info? Should we add to gitignore once it is an example file?

eparis replied:
So I wanted to put real(ish) data here as an example.  Also, I thought the .gitignore trick worked, but it doesn't...

Another option would be to make this file be .gitignore and make a openstack_config.yml.example. And the end user would have to cp the .example to just .yml and then make edits....

I'd still probably leave this data for reference though...

jeffbean replied:
OK just wondering. looks real alight :smile:. that password...

As for the file I haven't played with this so if vagrant tried to pick up this config by default the end-user might not understand why. I would say having a `.example` file extension would make it very clear. Then just call out the copy  in the README.

eparis replied:
I do use crappy passwords, but THAT one isn't one of my crappy passwords   :)

I'll move to the .example version and make the .yml in .gitignore.

jeffbean commented:
OK with my commits!

jeffbean commented:
Overall LGTM

jayunit100 commented:
ok w/ my commits also !

eparis committed:
Ansible: fast update of binaries

If you are using locally build binaries as a developer you likely will
want to just push those binaries to an existing cluster, not rerun the
entire playbook.  Add a tag to do just that.

eparis committed:
Ansible: convert DNS from kube_config_dir to kube_addons_dir

eparis committed:
Ansible: Add kube-ui kube-addon support

eparis committed:
Ansible: Vagrant: allow passing ansible tags to vagrant provision

Creating a cluster from scratch takes about 7 minutes. But if you just
rebuild the binaries and want to update those you don't want to have to
rerun the entire thing. There is an ansible tag 'binary-update' which
will do that. Now one can do
```
ANSIBLE_TAGS=binary-update vagrant provision
```
And it will push the new binaries.

eparis committed:
Ansible: start to support debian

jayunit100 commented:
Maybe we can merge this soon as wip and iterate. As of now I think it's useful to people?

eparis commented:
it's totally G2G from my PoV

jayunit100 commented:
For the record, this is what the right URLs for this looklike.  Would be good to have this in the vagrantfile, just for sanity checking.
```

        os.username     = ENV['OS_USERNAME']
        os.password     = ENV['OS_PASSWORD']
        os.flavor       = "m1.small"
        os.image = "Fedora 22 Cloud Base x86_64 (final)"
        os.openstack_auth_url     = "http://os1-public.osop.rhcloud.com:5000/v2.0/tokens/"
        os.security_groups    = ['default','newgroup']
        os.openstack_compute_url = "http://os1-public.osop.rhcloud.com:8774/v2/abcdabcdabcd"
        os.server_name = name 
        os.tenant_name = "ENG 1234"
        os.keypair_name = "JPeerindex"
        os.region = "OS1Public"
        os.floating_ip_pool   = 'os1_public'
        #os.floating_ip = "auto"
```

eparis commented:
@jayunit100 I did away with ENV all together.

https://github.com/eparis/kubernetes/blob/total-ansible/contrib/ansible/vagrant/openstack_config.yml.example

jayunit100 commented:
oh duh ok.  well LGTM for sure, ill run quick test now.

jayunit100 commented:
@eparis   i think one minor mod you might want to push before commiting...

```
os_auth_url: "http://os1-public.osop.rhcloud.com:5000/v2.0/"
``` should be 
```
os_auth_url: "http://os1-public.osop.rhcloud.com:5000/v2.0/tokens/"
```
with that change, vagrant up -> spins up VMS+ ansible provisioning  fully automated.  

So definetly ready to merge either way, but suggest updating the URL for os_auth_url

eparis commented:
porque?  I didn't seem to need the  `tokens/` and it wasn't included anywhere on the security page in the openstack instance I looked at. (although admittedly I agree it works)   Did you have a failure without the `tokens/`?

jayunit100 commented:
Yes, iirc I had to add tokens/.
 I guess we can settle for a comment above that line with the alternate tokens/ endpoint until we work out the exact logic

> On Jul 18, 2015, at 4:43 PM, Eric Paris <notifications@github.com> wrote:
> 
> porque? I didn't seem to need the tokens/ and it wasn't included anywhere on the security page in the openstack instance I looked at. (although admittedly I agree it works) Did you have a failure without the tokens/?
> 
> —
> Reply to this email directly or view it on GitHub.
>

jeffbean commented:
@eparis Anything more we need here to merge?

eparis commented:
I do not believe the jenkin's failure.  This stay entirely inside contrib/ansible.  If jenkins broke, it was a flake.

mikedanese commented:
@k8s-bot ok to test

eparis committed:
Install openssl when using make-ca-certs.sh

easyrsa requires openssl. So make sure it is installed.

jayunit100 committed:
Ansible: Vagrant: kubernetes openstack deployer

vbox deployer separated and deprecated. Readded-later

eparis commented:
@k8s-bot test this please

mikedanese commented:
Change was isolated to contrib/ansible/ and passing mungedocs and boilerplate checks

jayunit100 commented:
Great! Now we have upstream support for ansible on openstack as well as local vms. Thanks for finishing this of eric and Jeff.
